package/wireguard/package

```diff
@@ -71,3 +71,11 @@ package() {
 configure() {
     depmod -a
 }
+
+postremove() {
+    modprobe -r wireguard
```

Comment:
Do we care if this fails due to the module being in use?

Reply:
Good point! I think it's fine if removal fails if the module is in use, as long as the error displayed makes this clear. However, it would probably be better if this were a preremove script, instead, so that the files are not removed in case of failure. (I'm assuming that a preremove script failing would stop removal.)

Reply:
I tried to test what would happen if I run `modprobe -r wireguard` while pinging a host through a wireguard interface. The packets just go unanswered, and `modprobe` does not throw any module in use errors. My guess is that the module is only in use while it is creating interfaces.

Reply:
(The reason I originally had the module removal in a postremove script was that it would prevent the following scenario:
- preremove script unloads the module
- some other script loads the module
- the module files are removed
which would incorrectly leave the module loaded. This seems very unlikely, preremove is probably fine)

Reply:
> I think it's fine if removal fails if the module is in use, as long as the error displayed makes this clear. 

In which case we should probably output a slightly nicer error about what's happening, and include the source error message.

> (I'm assuming that a preremove script failing would stop removal.)

I think there is a bug where it doesn't?

> I tried to test what would happen if I run modprobe -r wireguard while pinging a host through a wireguard interface. The packets just go unanswered, and modprobe does not throw any module in use errors. My guess is that the module is only in use while it is creating interfaces.

![image](https://user-images.githubusercontent.com/1176979/104112085-aec38f80-52a7-11eb-9ace-4a7d62578634.png)
Looks like it fails silently :(

Reply:
Did you get the "module in use" error by just having another terminal with `ping` going? I couldn't reproduce your output.

I was hoping that `modprobe -r` would have different exit codes for different errors, so that we could have it only fail if the module is in use, but [it doesn't look like it](https://git.kernel.org/pub/scm/utils/kernel/kmod/kmod.git/tree/tools/modprobe.c#n954). Looking at the source code, I'm not sure at all why modprobe is failing silently. Oh well, I'll keep looking :)

Reply:
I ran client.sh and left everything configured.

Reply:
Hmm, I still can't reproduce that behavior. I can get a "module in use" error if I try to remove `udp_tunnel` because it's a dependency of wireguard. But I can't seem to get a "module in use" error for wireguard itself, no matter what I try. (My `modprobe --version` is
```
kmod version 25
-XZ +ZLIB -EXPERIMENTAL
```
Perhaps reMarkable 1 has a different version? Maybe this is a matter of differently configured kernels?)

I think our best option is to replace the current postremove `modprobe -r` with a preremove and preupgrade of 
```
rmmod wireguard
rmmod udp_tunnel || true
rmmod ip6_udp_tunnel || true
```
(The `|| true` is because we don't care if the dependencies fail to remove due to being used by other modules.)

Does this seem reasonable? Is rmmod's message of "Module wireguard is in use" clear enough? This feels like it would be a common problem for packaging kernel modules, but maybe everyone else just uses DKMS.

Reply:
rM1 has the same version and flags of `modprobe`.

Usually I'd say that it's a good enough message, but since this is a package and it's an error expect might happen we should give a more verbose explanation of what's going on. Maybe something like:
```
Failed to unload existing wireguard module as it is in use.
Likely you have a wireguard connection configured and will need to manually remove it before running the following:

rmmod wireguard udp_tunnel ip6_udp_tunnel
depmod -a

If it continues to error you may want to reboot your device.

```

Reply:
I just pushed a version with a more detailed error message. Sorry it's taken me so long to get back to you. I really appreciate you taking the time to think through edge cases and make this package better.

In the error message, I recommended using `opkg remove` (which will call the `modprobe -r` in the preremove script) instead of `rmmod`, but if you've gotten it to work with the latter but not the former, I'm happy to recommend `rmmod` instead. (I just thought it would be simpler for the user.) 

I put depmod in a postremove, to get wireguard out of the modules.dep file. It's called in the configure script, so I don't think that calling it manually would help with removing the module after a module in use error.

Reply:
@Eeems No rush on reviewing this, but I wanted to make sure you didn't miss it :)

Reply:
![image](https://user-images.githubusercontent.com/1176979/105243901-f01c3080-5b2c-11eb-858c-b1879892212d.png)
Hmm, I guess maybe we shouldn't try to remove and just let the user know they'll have to reboot for it to be fully removed?

Reply:
I think that's a good idea. Let me know if the postremove/postupgrade message I added could use more details/explanation.